apps/api/src/app/events/usecases/cancel-delayed/cancel-delayed.usecase.ts

```diff
@@ -34,6 +37,71 @@ export class CancelDelayed {
       }
     );
 
+    const mainDigestJob = transactionJobs.find((job) => isMainDigest(job.type, job.status));
+
+    if (!mainDigestJob) {
+      return true;
+    }
+
+    return await this.scheduleNextDigestJob(mainDigestJob);
+  }
+
+  private async scheduleNextDigestJob(job: PartialJob) {
+    const mainFollowerDigestJob = await this.jobRepository.findOne(
+      {
+        _mergedDigestId: job._id,
+        status: JobStatusEnum.MERGED,
+        type: StepTypeEnum.DIGEST,
+        _environmentId: job._environmentId,
+        _subscriberId: job._subscriberId,
+      },
+      '',
+      {
+        query: { sort: { createdAt: 1 } },
+      }
+    );
+
+    // meaning that only one trigger was send, and it was cancelled in the CancelDelayed.execute
```

Comment:
Not sure if keep those comments in the code or not, because IMO this flow is very complicated to grasp.